processor/src/chiplets/mod.rs

```diff
@@ -423,35 +460,23 @@ pub(super) enum ChipletsLookup {
     RequestAndResponse((usize, usize)),
 }
 
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
-#[allow(dead_code)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub(super) enum ChipletsLookupRow {
-    Hasher(HasherLookupRow),
+    Hasher(HasherLookup),
+    HasherMulti(Vec<HasherLookup>),
     Bitwise(BitwiseLookup),
     Memory(MemoryLookup),
 }
```

Comment:
Yep, I agree with both! I've moved it and added a performance-related TODO. I think can definitely find a more performant approach, but it will probably add considerable complexity, so it's better to have a baseline working & revisit.